### ISSUE TITLE
[@typescript-eslint/typedef] broken when used with a default parameter

### ISSUE THREAD

**Issue by octogonz** [typescript-eslint/typescript-eslint#784]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/typedef": [
      "error",
      {
        "arrayDestructuring": false,
        "arrowParameter": true,
        "memberVariableDeclaration": true,
        "parameter": true,
        "objectDestructuring": false,
        "propertyDeclaration": true,
        "variableDeclaration": true
      },
    ],
  }
}
```

```TS
// error @typescript-eslint/typedef: expected a type annotation
function f(x: number = 1): void {
}
```

**Expected Result**

The error should not be reported because there is a type annotation: `: number`

**Actual Result**

The error is reported.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0-alpha.4` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |

@JoshuaKGoldberg

**Comment by octogonz** [comment]
Here's a PR: https://github.com/typescript-eslint/typescript-eslint/pull/785
